lib/event_store_client/errors.rb

```diff
@@ -0,0 +1,95 @@
+# frozen_string_literal: true
+
+module EventStoreClient
+  class Error < StandardError
+    # @return [Hash]
+    def as_json(*)
+      to_h.transform_keys(&:to_s)
+    end
+
+    # @return [Hash]
+    def to_h
+      hash =
+        instance_variables.each_with_object({}) do |var, result|
+          key = var.to_s
+          key[0] = '' # remove @ sign
+          result[key.to_sym] = instance_variable_get(var)
+        end
+      hash[:message] = message
+      hash[:backtrace] = backtrace
+      hash
+    end
+  end
+
+  class StreamNotFoundError < Error
+    attr_reader :stream_name
+
+    # @param stream_name [String]
+    def initialize(stream_name)
+      @stream_name = stream_name
+      super("Stream #{stream_name.inspect} does not exist.")
+    end
+  end
+
+  class WrongExpectedVersionError < Error
+    attr_reader :wrong_expected_version, :caused_by
+
+    # @param wrong_expected_version [EventStore::Client::Streams::AppendResp::WrongExpectedVersion]
+    # @param caused_by [EventStoreClient::DeserializedEvent] an event on which
+    #   WrongExpectedVersionError error happened. It can be useful when appending array of events -
+    #   based on it you will know which events were appended and which weren't.
+    def initialize(wrong_expected_version, caused_by:)
+      @wrong_expected_version = wrong_expected_version
+      @caused_by = caused_by
```

Comment:
I would keep `extra` hash were we keep all additional info abut the error as it would be transferred automatically to the sentry. 
We can discuss it. 
At some point we are doing Sentry.capture_message(e.message, extra: e.extra || {})
https://github.com/yousty/yousty-api/blob/7a55d361d2ac036d269dfcfadfb115e078ad1641/spec/dummy/spec/yousty/error_handler_spec.rb

Reply:
The implementation is not meant to be compatible with yousty internals.

Reply:
But why not benefit auto-push to the Sentry if we slightly change?

Reply:
Should we also take care about another services? :)